test/tests/functional/pbs_nonprint_characters.py

```diff
@@ -271,16 +280,22 @@ def test_nonprint_character_default_qsubV(self):
                                 {'default_qsub_arguments': '-V'})
             script = ['sleep 5']
             script += ['env | grep NONPRINT_VAR']
-            j = Job(self.du.get_current_user())
+            j = Job(TEST_USER)
             j.create_script(body=script)
-            jid = self.server.submit(j)
+            exp = "export NONPRINT_VAR=\"X%sY\"" % ch
+            file_n = j.create_script(body=script)
+            script = [exp]
+            script += ['/opt/pbs/bin/qsub %s' % file_n]
+            j.create_script(body=script)
```

Comment:
I'm not sure I understand this.  Why are you calling j.create_script() twice?  Won't the second call overwrite the first?

Reply:
Yes, first script is copied to remote host & is being called by second script.
First script contains - script1
sleep 5
env | grep NONPRINT_VAR

second script - 
export NONPRINT_VAR=value
/opt/pbs/bin/qsub script1
So, that environment variable is set in same session

Reply:
As with my above comment, use du.create_temp_file()